examples/trials/mnist/mnist_before.py

```diff
@@ -0,0 +1,219 @@
+                               self.keep_prob: 1.0})
+                logger.debug('test accuracy %g', test_acc)
+    def evaluate(self, mnist):
+        test_acc = self.accuracy.eval(
+            feed_dict={self.images: mnist.test.images,
+                       self.labels: mnist.test.labels,
+                       self.keep_prob: 1.0})
+        logger.debug('Final result is %g', test_acc)
+        logger.debug('Send final result done.')
+        return test_acc
+
+
+def conv2d(x_input, w_matrix):
+    """conv2d returns a 2d convolution layer with full stride."""
+    return tf.nn.conv2d(x_input, w_matrix, strides=[1, 1, 1, 1], padding='SAME')
+
+
+def max_pool(x_input, pool_size):
+    """max_pool downsamples a feature map by 2X."""
+    return tf.nn.max_pool(x_input, ksize=[1, pool_size, pool_size, 1],
+                          strides=[1, pool_size, pool_size, 1], padding='SAME')
+
+
+def weight_variable(shape):
+    """weight_variable generates a weight variable of a given shape."""
+    initial = tf.truncated_normal(shape, stddev=0.1)
+    return tf.Variable(initial)
+
+
+def bias_variable(shape):
+    """bias_variable generates a bias variable of a given shape."""
+    initial = tf.constant(0.1, shape=shape)
+    return tf.Variable(initial)
+
+def write_log():
+    # Write log
+    graph_location = tempfile.mkdtemp()
+    logger.debug('Saving graph to: %s', graph_location)
+    train_writer = tf.summary.FileWriter(graph_location)
+    train_writer.add_graph(tf.get_default_graph())     
+
+def run_trial(params):
+    '''
+    Main function, build mnist network, run and send result to NNI.
+    '''
+    # Import data
+    mnist = input_data.read_data_sets(params['data_dir'], one_hot=True)
+    print('Mnist download data down.')
+    logger.debug('Mnist download data down.')
+
+    # Create the model
+    # Build the graph for the deep net
+    mnist_network = MnistNetwork(channel_1_num=params['channel_1_num'],
+                                 channel_2_num=params['channel_2_num'],
+                                 conv_size=params['conv_size'],
+                                 hidden_size=params['hidden_size'],
+                                 pool_size=params['pool_size'],
+                                 learning_rate=params['learning_rate'])
+    mnist_network.build_network()
+    logger.debug('Mnist build network done.')
+    write_log()
+
+    test_acc = 0.0
+    with tf.Session() as sess:
+        mnist_network.train(sess, mnist)
+        test_acc = mnist_network.evaluate(mnist)
+        print ("Final result is", test_acc)
+
+if __name__ == '__main__':
+    try:
+        params = {
+            'data_dir': '/tmp/tensorflow/mnist/input_data',
+            'dropout_rate': 0.5,
+            'channel_1_num': 32,
+            'channel_2_num': 64,
+            'conv_size': 5,
+            'pool_size': 2,
+            'hidden_size': 1024,
+            'learning_rate': 1e-4,
```

Comment:
There are some updates on mnist example recently,  could you sync from master mnist example again?

Reply:
see https://github.com/Microsoft/nni/pull/648/commits/d6bbb79c92f706711b7d42c65b0af790a8415b8e